test/libsyncengine/requests/testserverrequests.cpp

```diff
@@ -0,0 +1,52 @@
+/*
+ * Infomaniak kDrive - Desktop
+ * Copyright (C) 2023-2024 Infomaniak Network SA
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include "testserverrequests.h"
+
+#include "libparms/db/parmsdb.h"
+#include "libsyncengine/requests/serverrequests.h"
+#include "requests/parameterscache.h"
+#include "test_utility/testhelpers.h"
+#include "utility/types.h"
+
+namespace KDC {
+
+void TestServerRequests::setUp() {
+    // Create parmsDb
+    bool alreadyExists = false;
+    std::filesystem::path parmsDbPath = Db::makeDbName(alreadyExists, true);
+    ParmsDb::instance(parmsDbPath, KDRIVE_VERSION_STRING, true, true);
+}
+
+void TestServerRequests::tearDown() {
+    ParmsDb::instance()->close();
+    ParmsDb::reset();
+}
+
+void TestServerRequests::testFixProxyConfig() {
+    ProxyConfig proxyConfig = ParametersCache::instance()->parameters().proxyConfig();
+    proxyConfig.setType(ProxyType::Undefined);
+    ParametersCache::instance()->parameters().setProxyConfig(proxyConfig);
+    CPPUNIT_ASSERT_EQUAL(ParametersCache::instance()->save(), ExitCode::Ok);
```

Comment:
The `CPPUNIT_ASSERT_EQUAL` doc says that `expected` is the first argument while `actual` is the second argument.